src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Helpers/AdalIdHelper.cs

```diff
@@ -67,6 +67,8 @@ internal static class AdalIdParameter
     /// </summary>
     internal static class AdalIdHelper
     {
+        private static readonly Regex AdalVersionRegex = new Regex(@"Version=[\d]+.[\d+]+.[\d]+.[\d]+", RegexOptions.CultureInvariant);
```

Comment:
You may want to think about RegexOptions.Compiled. The first check is slower, but the future are much faster.
https://msdn.microsoft.com/en-us/library/system.text.regularexpressions.regexoptions(v=vs.110).aspx

Reply:
I like this, but looks like netstandard1.1 doesn't support RegexOptions.Compiled